### PR TITLE
ENH: Column Vertical Profile (CVP) Addition

### PR DESCRIPTION
**Pull request by jrobrien91**
Based on Murphy et al. (2020) and Bukovčić et al. (2020), a function to retrieve a columnar vertical profile (CVP) was added to the columnsect.py utility. 

Instead of extracting single gates from a single azimuth above a location (e.g. `get_field_location`), `column_vertical_profile` allows the user to define the number of azimuths and range gates to include with in this extraction. 

While providing similar functionality as `get_field_location`, `column_vertical_profile` adds more samples to the extracted column, increasing frequency of valid data at lower elevation gates. This ultimately allows for easier comparison with in-situ observations. 

- [X] Tests added
- [X] Documentation reflects changes

### CODE REVIEW

**Comment by zssherman** [comment]
@jrobrien91 Thanks for the PR, I'll review here in a moment!

File: pyart/util/columnsect.py [hunk 1]
```diff
@@ -11,6 +11,159 @@
...
+    sphere_distance
+    for_azimuth
+    get_sweep_rays
+    subset_fields
+    assemble_column
+
+    Returns
+    -------
+    column : xarray
+        Xarary Dataset containing the radar column above the target for
```
Comment (zssherman) [inline pyart/util/columnsect.py:48]: Xarary to Xarray

File: pyart/util/columnsect.py [hunk 2]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+    Extract the specific rays for a given azimuth from a radar sweep
+
+    Azimuth spread determines the +/- degrees azimuth to include within
+    the extraction by multipling the azimuth resolution by input value.
+
+    Parameters
+    ----------
+    radar_sweep : pyart.core.radar object
+        Radar Sweep from which the rays are extracted from
+
```
Comment (zssherman) [inline pyart/util/columnsect.py:494]: Can remove spacing between parameter list

File: pyart/util/columnsect.py [hunk 3]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+    azimuth : float [degrees]
+        Forward Azimuth Angle from Radar to Target in Degreees
+
```
Comment (zssherman) [inline pyart/util/columnsect.py:497]: Same here

File: pyart/util/columnsect.py [hunk 4]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+    Azimuth_Spread : int
+        Number of azimuth angles to include within extraction list
+
+    Returns
+    -------
+    center_rays : list [integers]
+        List of integers cooresponding to ray indices within the azimuth
+        directly over the target
+
```
Comment (zssherman) [inline pyart/util/columnsect.py:506]: Remove space

File: pyart/util/columnsect.py [hunk 5]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+    return centerline, spread
+
+
+def subset_fields(radar, ray, target_gates):
+    """
+    Parameter
+    ---------
+    radar : pyart.core.radar object
+    Radar Sweep from which fields are extracted from the target locations
```
Comment (zssherman) [inline pyart/util/columnsect.py:528]: Indent by 4 spaces

File: pyart/util/columnsect.py [hunk 6]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+
+    target_gates : list
+    List containing indices for the gates of interest
```
Comment (zssherman) [inline pyart/util/columnsect.py:531]: Same here, indent

File: pyart/util/columnsect.py [hunk 7]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+
+    Returns
+    -------
+    fields : dict
+    dictionary containing averaged subset fields for target location
```
Comment (zssherman) [inline pyart/util/columnsect.py:536]: Indent here as well

File: pyart/util/columnsect.py [hunk 8]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+    With a dictionary containing the extracted fields from a radar sweep,
+    assemble individual gates and fields into an xarray DataSet
+
+    Parameters
+    ----------
+    total_moment : dict
+        Dictionary containing the extracted fields from the radar object.
+        File requires at least the height of the individual gates and
+        the start time of the volumetric scan.
+
```
Comment (zssherman) [inline pyart/util/columnsect.py:567]: Remove space

File: pyart/util/columnsect.py [hunk 9]
```diff
@@ -326,6 +480,207 @@ def get_column_rays(radar, azimuth):
...
+    azimuth : float, [degrees]
+        azimuth angle from the radar where
+        target is located within the scan.
+        output is in degrees.
+
```
Comment (zssherman) [inline pyart/util/columnsect.py:572]: Remove space

Comment (zssherman) [reply pyart/util/columnsect.py:572]: More below

**Review by zssherman — CHANGES_REQUESTED** [review]
@jrobrien91 Mostly docstring fixes etc. The code itself looks good. The new unit test looks to be failing

**Comment by jrobrien91** [comment]
@zssherman thanks for catching all those! I've resubmitted the changes

**Review by zssherman — APPROVED** [review]

**Comment by zssherman** [comment]
@jrobrien91 I did a few more changes, not sure why the doc build is failing though..

**Comment by jrobrien91** [comment]
@zssherman thanks for catching those doc-string issues! Not sure why the docs are failing either. 

looks like xradar issue?

**Comment by zssherman** [comment]
@jrobrien91 Yeah we might need to wait on @mgrover1 , I can dig into the code in the meantime

**Comment by zssherman** [comment]
@jrobrien91 Actually looks more like a pathlib issue, i'll check their forums

**Comment by zssherman** [comment]
Pinning 3.11 for doc build for now.

**Review by mgrover1 — APPROVED** [review]
This looks great! Nice work @jrobrien91

File: doc/environment.yml [hunk 10]
```diff
@@ -3,6 +3,7 @@ channels:
   - conda-forge
   - defaults
 dependencies:
+  - python=3.11
```
Comment (mgrover1) [inline doc/environment.yml:6]:
```suggestion
```

Comment (mgrover1) [reply doc/environment.yml:6]: We do not need this anymore now that we have a release of datatree with the fix.

**Review by zssherman — APPROVED** [review]

**Review by mgrover1 — APPROVED** [review]

**Comment by mgrover1** [comment]
The CI/linting issues have been resolved! Good to merge!

**Pull request merged**